src/main/java/org/apache/cassandra/sidecar/config/yaml/ServiceConfigurationImpl.java

```diff
@@ -88,117 +107,45 @@ public class ServiceConfigurationImpl implements ServiceConfiguration
-             throttleConfiguration,
-             ssTableUploadConfiguration,
-             new SSTableImportConfigurationImpl(),
-             DEFAULT_WORKER_POOLS_CONFIGURATION,
-             jmxConfiguration);
-    }
-
-    public ServiceConfigurationImpl(int requestIdleTimeoutMillis,
-                                    long requestTimeoutMillis,
-                                    SSTableUploadConfiguration ssTableUploadConfiguration)
-    {
+    @JsonProperty(value = TRAFFIC_SHAPING_PROPERTY)
+    protected final TrafficShapingConfiguration trafficShapingConfiguration;
 
-        this(DEFAULT_HOST,
-             DEFAULT_PORT,
-             requestIdleTimeoutMillis,
-             requestTimeoutMillis,
-             DEFAULT_ALLOWABLE_SKEW_IN_MINUTES,
-             new ThrottleConfigurationImpl(),
-             ssTableUploadConfiguration,
-             new SSTableImportConfigurationImpl(),
-             DEFAULT_WORKER_POOLS_CONFIGURATION,
-             new JmxConfigurationImpl());
-    }
-
-    public ServiceConfigurationImpl(String host,
-                                    int port,
-                                    int requestIdleTimeoutMillis,
-                                    long requestTimeoutMillis,
-                                    int allowableSkewInMinutes,
-                                    ThrottleConfiguration throttleConfiguration,
-                                    SSTableUploadConfiguration ssTableUploadConfiguration,
-                                    SSTableImportConfiguration ssTableImportConfiguration,
-                                    Map<String, ? extends WorkerPoolConfiguration> workerPoolsConfiguration,
-                                    JmxConfiguration jmxConfiguration)
+    /**
+     * Constructs a new {@link ServiceConfigurationImpl} with the default values
+     */
+    public ServiceConfigurationImpl()
     {
-        this.host = host;
-        this.port = port;
-        this.requestIdleTimeoutMillis = requestIdleTimeoutMillis;
-        this.requestTimeoutMillis = requestTimeoutMillis;
-        this.allowableSkewInMinutes = allowableSkewInMinutes;
-        this.throttleConfiguration = throttleConfiguration;
-        this.ssTableUploadConfiguration = ssTableUploadConfiguration;
-        this.ssTableImportConfiguration = ssTableImportConfiguration;
-        this.jmxConfiguration = jmxConfiguration;
-        if (workerPoolsConfiguration == null || workerPoolsConfiguration.isEmpty())
-        {
-            this.workerPoolsConfiguration = DEFAULT_WORKER_POOLS_CONFIGURATION;
-        }
-        else
-        {
-            this.workerPoolsConfiguration = workerPoolsConfiguration;
-        }
+        this(builder());
     }
 
-    public ServiceConfigurationImpl(String host)
+    /**
+     * Constructs a new {@link ServiceConfigurationImpl} with the configured {@link Builder}
+     *
+     * @param builder the builder object
+     */
+    protected ServiceConfigurationImpl(Builder builder)
```

Comment:
this is another case of exploding number of constructors. So I decided to simplify with a builder